Makefile

```
@@ -112,18 +122,18 @@ modules: $(OCRD_MODULES)
 ifneq (,$(wildcard .git))
 $(OCRD_MODULES): always-update
 ifneq ($(NO_UPDATE),1)
-	sem --fg --id ocrd_all_git git submodule sync $(GIT_RECURSIVE) $@
+	$(SEM) git submodule sync $(GIT_RECURSIVE) $@
 	if git submodule status $(GIT_RECURSIVE) $@ | grep -qv '^ '; then \
-		sem --fg --id ocrd_all_git git submodule update --init $(GIT_RECURSIVE) $@ && \
+		$(SEM) git submodule update --init $(GIT_RECURSIVE) $@ && \
 		touch $@; fi
 endif
 endif
 
 deinit: clean
 .PHONY: deinit
 deinit:
-	git submodule deinit --all # --force
-	git submodule status | while read stat dir ver; do rmdir $$dir; done
+	-git submodule deinit --all # --force
```

Why do you want to ignore errors here? Wouldn't it be better to fail if the `deinit`target cannot deinit all submodules?

---

Because not all modules might be checked out (initialized) before. git will still return an error in that case.

---

I cannot confirm that. Deinit only fails for me when a submodule has modified contents, and for that case that's the desired behaviour.

---

Yeah, I agree `deinit` should fail under these conditions.